### DISCUSSION TITLE
How to add three wheeler in ODC

### DISCUSSION THREAD

**Discussion by nishantsah08** [opendatacam/opendatacam#552] [Q&A]
It seems for the vehicles ODC supports only two wheeler and four wheeler. How do I add a new object for detection?
Do I have to retrain the yolo algorithm ?
Any detailed steps would be much appreciated?
I am trying to add support for auto and toto(three wheeler, Indian context)

**Comment by vsaw** [comment]
You would have to train your own custom model and replace the standard Yolov4 model used by OpenDataCam. 

Darknet has a section on how to train your model but there are also plenty of tutorials available https://github.com/opendatacam/darknet#how-to-train-to-detect-your-custom-objects

Once you have your model you can configure ODC to use it like described here https://github.com/opendatacam/opendatacam/blob/master/documentation/CONFIG.md#use-custom-neural-network-weights

**Comment by nishantsah08** [comment]
Once model is trained on custom dataset does it have to be optimized for  tensorrt or it can be simply plugged it in?

**Reply by vsaw** [reply]
Tensorrt? ODC uses Darknet. Once you have finished training you can use it directly. Just need to change the config to point to your weights instead of the default Yolo weights.
